src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataDeltaFeedSerializer.cs

```diff
@@ -459,11 +515,20 @@ public virtual void WriteDeltaDeletedEntry(object graph, ODataWriter writer, ODa
         /// <param name="writeContext">The <see cref="ODataSerializerContext"/>.</param>
         public virtual async Task WriteDeltaDeletedEntryAsync(object graph, ODataWriter writer, ODataSerializerContext writeContext)
         {
-            ODataDeletedResource deletedResource = GetDeletedResource(graph);
-            if (deletedResource != null)
+            ODataResourceSerializer serializer = SerializerProvider.GetEdmTypeSerializer(_elementType) as ODataResourceSerializer;
+            ResourceContext resourceContext = serializer.GetResourceContext(graph, writeContext);
+            SelectExpandNode selectExpandNode = serializer.CreateSelectExpandNode(resourceContext);
+
+            if (selectExpandNode != null)
             {
-                await writer.WriteStartAsync(deletedResource);
-                await writer.WriteEndAsync();
+                ODataDeletedResource deletedResource = GetDeletedResource(graph, resourceContext, serializer, selectExpandNode, writeContext.IsUntyped);
+
+                if (deletedResource != null)
+                {
+                    await writer.WriteStartAsync(deletedResource);
+                    serializer.WriteDeltaComplexProperties(selectExpandNode, resourceContext, writer);
```

Comment:
Does this method perform any I/O operation? If it does, shouldn't it be async?

Reply:
I have called the asynchronous version of the method.